src/olympia/files/migrations/0014_auto_20210824_1722.py

```diff
@@ -0,0 +1,20 @@
+# Generated by Django 3.2.6 on 2021-08-24 17:22
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('versions', '0016_auto_20210325_1320'),
+        ('files', '0013_auto_20210830_1424'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='file',
+            name='version',
+            field=models.OneToOneField(on_delete=django.db.models.deletion.CASCADE, to='versions.version'),
+        ),
+    ]
```

Comment:
This generates:
```sql
ALTER TABLE `files` DROP FOREIGN KEY `files_version_id_44ed234d_fk_versions_id`;
DROP INDEX `files_version_id_44ed234d_fk_versions_id` ON `files`;
ALTER TABLE `files` ADD CONSTRAINT `files_version_id_44ed234d_uniq` UNIQUE (`version_id`);
ALTER TABLE `files` ADD CONSTRAINT `files_version_id_44ed234d_fk_versions_id` FOREIGN KEY (`version_id`) REFERENCES `versions` (`id`);
```
which seems partially redundant? - only adding the `UNIQUE` constraint is necessary?

Reply:
That's a bit annoying... I'm guessing django has no specific migration path from `ForeignKey` to `OneToOneField` so it has to drop everything to re-create... You could consider doing the relevant operation in a `RunSQL` with a `state_operations=[migrations.AlterField(...)]` to avoid this ?

Reply:
btw, I just double checked the sql statements don't do anything else by running locally and comparing the outputs of `show create table files` before and after.  
Confirming that the end result is
```KEY `files_version_id_44ed234d_fk_versions_id` (`version_id`)``` being replaced with 
```UNIQUE KEY `files_version_id_44ed234d_uniq` (`version_id`)```